TreesAndGraphs/isBalanced/solution1.js

```diff
@@ -0,0 +1,75 @@
+// is the binary tree balanced? -> return true
+
+class Node {
+    constructor(value, left, right) {
+        this.value = value;
+        this.right = right || null;
+        this.left = left || null;
+    }
+}
+
+// Time complexity is Big O(N);
+// Space complexity is Big O(log N) because we are building a stack of each subtree;
+function isBalanced(root) {
+    let min;
+    let max;
+
+    function helper(node) {
+        if (node == null) {
+            return {
+                min: 0,
+                max: 0
+            }
+        } else {
+            let left_obj = helper(node.left);
+            let right_obj = helper(node.right);
+            min = Math.min(left_obj.min, right_obj.min) + 1;
+            max = Math.max(left_obj.max, right_obj.max) + 1;
+        }
+        return {
+            min,
+            max
+        }
+    }
+
+    let max_and_min = helper(root);
+    if (Math.abs(max_and_min.max - max_and_min.min) > 1) {
+        return false
+    } else {
+        return true;
+    }
+}
+
+let tree1 = new Node(10,
+    new Node(3),
+    new Node(25)
+);
+
+let tree3 = new Node(5,
+    new Node(3,
+        null,
+        new Node(4)
+    ),
+    new Node(15,
+        new Node(13),
+        new Node(25,
+            null,
+            new Node(31)
+        )
+    )
+);
+
+let tree5 = new Node(10,
+    new Node(7),
+    new Node(15,
+        new Node(13),
+        new Node(25,
+            null,
+            new Node(31)
+        )
+    )
+);
+
+console.log(isBalanced(tree1)); //true
+console.log(isBalanced(tree5)); //false
+console.log(isBalanced(tree3)); //false
```

Comment:
@annamikmorozova This is new to me, so I had to do a bit of research, and I could definitely be wrong here, but isn't tree3 balanced? 
![aww-board (3)](https://user-images.githubusercontent.com/34624364/90147748-86635380-dd50-11ea-84b1-bd5e0f6a1cdb.png)

Reply:
I think row 3 has to be filled in order for it to be balanced. And row 4 does not have to be complete

Reply:
https://www.youtube.com/watch?v=LU4fGD-fgJQ